src/encoded/static/libs/bootstrap/tooltip.js

```diff
@@ -0,0 +1,120 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+
+// This module lets you have a tooltip pop up when any component is hovered over or focused. The
+// general usage is:
+//
+// <Tooltip trigger={<Trigger />} tooltipId="html-id-unique-on-page" css="custom-css-classes">
+//     <TooltipContent />
+// </Tooltip>
+//
+// The component that the user can hover over to trigger the tooltip is given in the `trigger`
+// property and appears in the DOM whereever you place the <Tooltip> component. Put the contents
+// of the tooltip between the opening and closing <Tooltip> tags. This can simply be a string if
+// that's all you need, or a full-fledged component.
+
+export default class Tooltip extends React.Component {
+    constructor() {
+        super();
+        this.state = {
+            tooltipDisplayed: false,
+        };
+        this.timer = null;
+        this.hoveringOverTooltip = false;
+        this.handleTriggerMouseEnter = this.handleTriggerMouseEnter.bind(this);
+        this.handleTriggerMouseLeave = this.handleTriggerMouseLeave.bind(this);
+        this.handleTooltipMouseEnter = this.handleTooltipMouseEnter.bind(this);
+        this.handleTooltipMouseLeave = this.handleTooltipMouseLeave.bind(this);
+        this.handleTriggerFocus = this.handleTriggerFocus.bind(this);
+        this.handleTriggerBlur = this.handleTriggerBlur.bind(this);
+    }
+
+    handleTriggerMouseEnter() {
+        // Mouse entered the tooltip trigger element.
+        this.setState({ tooltipDisplayed: true });
+
+        // If we happen to have a running timer, clear it so we don't hide the tooltip while
+        // hovering over the trigger.
+        if (this.timer) {
+            clearTimeout(this.timer);
+            this.timer = null;
+        }
+    }
+
+    handleTriggerMouseLeave() {
+        // Start a timer that might hide the tooltip after a second passes. It won't hide the
+        // tooltip if they're now hovering over the tooltip itself.
+        this.timer = setTimeout(() => {
+            this.timer = null;
+            if (!this.hoveringOverTooltip) {
+                this.setState({ tooltipDisplayed: false });
+            }
+        }, 1000);
+    }
+
+    handleTriggerFocus() {
+        // Show tooltip when keyboard focus lands on tooltip trigger.
```

Comment:
Personal Preference Comment: This seems like a comment that describes the function. It may be advantageous to use JSDoc comment style and have this on top of the method

Reply:
If you have something to describe official ways to support JSDoc in methods, I’d like to see it. Possibly not for this branch though as this is so high priority.

Reply:
similar to how they are used on classes... http://usejsdoc.org/tags-function.html? Again, this no big deal and the least of our worries.